# PID values are positive for forward acting PID loops and negative for
# reverse acting PID loops. P, I, and D should all be the same sign.
throttle_p: 4.0
throttle_i: 2.5
throttle_d: 0.0
throttle_accumulator_enable_threshold: 10.0
throttle_accumulator_max: 5.0
throttle_accumulator_min: -5.0

# Data derived from:
# https://wiki.bitcraze.io/misc:investigations:thrust
thrust_model:
    thrust_scale_factor: 0.409 # mass (kg) / g / num_props
    A_ge: 0.0
    d0: 1.0
    voltage_polynomial: [1.0]
    throttle_b: 0.34352
    throttle_c: -0.24205

# Thrust levels are in m/s^2
min_thrust: 7.0
max_thrust: 100.0

battery_timeout: 0.5

pitch_p: 0.15
pitch_i: 0.075
pitch_d: 0.0
pitch_accumulator_max: 10.0
pitch_accumulator_min: -10.0
pitch_accumulator_enable_threshold: 10.0

roll_p: 0.15
roll_i: 0.075
roll_d: 0.0
roll_accumulator_max: 10.0
roll_accumulator_min: -10.0
roll_accumulator_enable_threshold: 10.0

throttle_max      : 1.0
throttle_min      : 0.0
throttle_max_rate : 200.0

pitch_max         : 0.175
pitch_min         : -0.175
pitch_max_rate    : 5.0

roll_max          : 0.175
roll_min          : -0.175
roll_max_rate     : 5.0

yaw_max           : 0.5
yaw_min           : -0.5
yaw_max_rate      : 10.0

update_frequency: 60.0
startup_timeout: 10.0
update_timeout: 0.2

# Rate at which to ramp the throttle during takeoff in percent a second
takeoff_throttle_ramp_rate: 0.1

# Rate at which to accelerate from 0 to descent velocity
# must be positive (m/s^2)
descend_acceleration: 0.1

# Rate at which to descent
# must be negative
descend_rate: -0.3

# How hard to brace
brace_impact_velocity: -0.3

# When to begin bracing
brace_impact_start_height: 0.15

# If we go above this height during brace low level motion considers
# this an emergency situation. The land sequence didn't work.
brace_impact_failure_height: 0.2

# This parameter will be obsolete when the foot touch sensors are implemented
brace_impact_success_height: 0.05

#height at which the contact switches must be unpressed
takeoff_max_height_switch_pressed: 0.25
